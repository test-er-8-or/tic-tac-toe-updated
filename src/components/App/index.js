import * as React from 'react'

import Board from '../Board'
import Square from '../../containers/Square'
import styled from 'styled-components'
import { times } from 'ramda'

const NUMBER_OF_SQUARES = 9

const StyledApp = styled.div`
  display: grid;
  font-family: 'Verdana', sans-serif;
  grid-template-areas: 'board';
  height: 100vh;
  margin: 0;
  padding: 0;
  width: 100vw;
`
StyledApp.defaultName = 'StyledApp'

export default function App () {
  return (
    <StyledApp>
      <Board>
        {times(
          square => (
            <Square key={square} index={square} />
          ),
          NUMBER_OF_SQUARES
        )}
      </Board>
    </StyledApp>
  )
}
