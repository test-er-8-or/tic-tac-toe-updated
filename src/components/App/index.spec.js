import * as React from 'react'

import App from './'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'

const middlewares = []
const mockStore = configureStore(middlewares)
const store = mockStore({ moves: [4, 0, 1] })

describe('components:App', () => {
  it('renders the App with a blank game board and nine squares', () => {
    expect(
      toJson(
        shallow(
          <Provider store={store}>
            <App />
          </Provider>
        )
          .first()
          .render()
      )
    ).toMatchSnapshot()
  })
})
