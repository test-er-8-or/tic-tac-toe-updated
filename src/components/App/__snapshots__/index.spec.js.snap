// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:App matches the snapshot 1`] = `
.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: grid;
  grid-area: board;
  grid-gap: 0;
  grid-template-areas: 'zero one two' 'three four five' 'six seven eight';
  grid-template-columns: 20vh 20vh 20vh;
  grid-template-rows: 20vh 20vh 20vh;
  height: 60vh;
  justify-self: center;
  margin: auto;
  width: 60vh;
}

.c2 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 0 2px 2px 0;
  color: hsla(6,59%,50%,1);
  cursor: default;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
}

.c3 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 0 2px 2px 0;
  color: hsla(145,63%,32%,1);
  cursor: default;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
}

.c4 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 0 0 2px 0;
  color: hsla(6,59%,50%,1);
  cursor: default;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
}

.c5 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 0 0 2px 0;
  color: hsla(145,63%,32%,1);
  cursor: default;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
}

.c6 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 0 2px 0 0;
  color: hsla(6,59%,50%,1);
  cursor: default;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
}

.c7 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 0 2px 0 0;
  color: hsla(145,63%,32%,1);
  cursor: default;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
}

.c8 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 0 0 0 0;
  color: hsla(6,59%,50%,1);
  cursor: default;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  text-align: center;
  text-transform: uppercase;
}

.c0 {
  display: grid;
  font-family: 'Verdana',sans-serif;
  grid-template-areas: 'board';
  height: 100vh;
  margin: 0;
  padding: 0;
  width: 100vw;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      x
    </div>
    <div
      class="c3"
    >
      o
    </div>
    <div
      class="c4"
    >
      x
    </div>
    <div
      class="c3"
    >
      o
    </div>
    <div
      class="c2"
    >
      x
    </div>
    <div
      class="c5"
    >
      o
    </div>
    <div
      class="c6"
    >
      x
    </div>
    <div
      class="c7"
    >
      o
    </div>
    <div
      class="c8"
    >
      x
    </div>
  </div>
</div>
`;
